### PR TITLE
[GNA] Fixed Transpose replacement with Gather

### PR DESCRIPTION
**Pull request by mryzhov**
### Details:
 - Fixed initial implementation and removed tail/untail checks

### Tickets:
 - *114979*

### CODE REVIEW

File: src/plugins/intel_gna/src/transformations/gather_sinking_transpose_reshape.cpp [hunk 1]
```diff
@@ -19,154 +19,120 @@
...
-    return std::make_pair(reshape_new, gather);
+    ov::replace_output_update_name(reshape->output(0), new_nodes.back()->output(0));
+    ov::copy_runtime_info({reshape}, new_nodes);
+
+    return new_nodes;
 }
 
-NodePair SinkBackward(NodePtr transpose, std::shared_ptr<Constant> transpose_constant, NodePtr reshape) {
-    const int64_t gather_axis_value = graph_utils::get_first_valuable_dim_id(reshape->get_input_shape(0));
+std::vector<std::shared_ptr<ov::Node>> SinkBackward(std::shared_ptr<ov::Node> reshape,
```
Comment (sirzabek) [inline src/plugins/intel_gna/src/transformations/gather_sinking_transpose_reshape.cpp:71]: Please use function naming convention here and in other places - lower case with underscore.

File: src/plugins/intel_gna/src/transformations/gather_sinking_transpose_reshape.cpp [hunk 2]
```diff
@@ -19,154 +19,120 @@
...
-    for (size_t n = 0; n < output_shape_4d[0]; ++n) {
-        for (size_t i = 0; i < output_shape_4d[1]; ++i) {
-            for (size_t j = 0; j < output_shape_4d[2]; ++j) {
-                for (size_t k = 0; k < output_shape_4d[3]; ++k) {
-                    gather_order.push_back(n * xyz[0] + i * xyz[1] + j * xyz[2] + k * xyz[3]);
-                }
-            }
+    auto transpose_const = ov::as_type_ptr<Constant>(transpose->get_input_node_shared_ptr(1));
+    const auto gather_indices_value =
+        graph_utils::make_gather_indexes_from_transpose_axes(transpose->get_input_shape(0),
```
Comment (sirzabek) [inline src/plugins/intel_gna/src/transformations/gather_sinking_transpose_reshape.cpp:34]: Please use indexes/indices consistently here and in other places.

Comment (mryzhov) [reply src/plugins/intel_gna/src/transformations/gather_sinking_transpose_reshape.cpp:34]: we agreed to use indexes, this is not a new function

Comment (mryzhov) [reply src/plugins/intel_gna/src/transformations/gather_sinking_transpose_reshape.cpp]: renamed

**Review by sirzabek — APPROVED** [review]

**Pull request merged**